src/wazuh_modules/syscollector/syscollector_bsd.c

```diff
@@ -1361,13 +1360,209 @@ int getGatewayList(OSHash *gateway_list){
+            break;
+        case TSI_S_SYN_SENT:
+            snprintf(port_state, STATE_LENGTH, "%s", "syn_sent");
+            break;
+        case TSI_S_SYN_RECEIVED:
+            snprintf(port_state, STATE_LENGTH, "%s", "syn_recv");
+            break;
+        case TSI_S_FIN_WAIT_1:
+            snprintf(port_state, STATE_LENGTH, "%s", "fin_wait1");
+            break;
+        case TSI_S_FIN_WAIT_2:
+            snprintf(port_state, STATE_LENGTH, "%s", "fin_wait2");
+            break;
+        case TSI_S_TIME_WAIT:
+            snprintf(port_state, STATE_LENGTH, "%s", "time_wait");
+            break;
+        case TSI_S_CLOSED:
+            snprintf(port_state, STATE_LENGTH, "%s", "close");
+            break;
+        case TSI_S__CLOSE_WAIT:
+            snprintf(port_state, STATE_LENGTH, "%s", "close_wait");
+            break;
+        case TSI_S_LAST_ACK:
+            snprintf(port_state, STATE_LENGTH, "%s", "last_ack");
+            break;
+        case TSI_S_LISTEN:
+            snprintf(port_state, STATE_LENGTH, "%s", "listening");
+            break;
+        case TSI_S_CLOSING:
+            snprintf(port_state, STATE_LENGTH, "%s", "closing");
+            break;
+        default:
+            snprintf(port_state, STATE_LENGTH, "%s", "unknown");
+            break;
+    }
+    return port_state;
+}
+
+void sys_ports_mac(int queue_fd, const char* WM_SYS_LOCATION, int check_all) {
+
+    time_t now = time(NULL);
     struct tm localtm;
+    localtime_r(&now, &localtm);
+
+    char *timestamp;
+    os_calloc(TIME_LENGTH, sizeof(char), timestamp);
+    snprintf(timestamp, TIME_LENGTH - 1, "%d/%02d/%02d %02d:%02d:%02d",
+            localtm.tm_year + 1900, localtm.tm_mon + 1,
+            localtm.tm_mday, localtm.tm_hour, localtm.tm_min, localtm.tm_sec);
+
     int random_id = os_random();
+    if (random_id < 0) {
+        random_id = -random_id;
+    }
+
+    // Define time to sleep between messages sent
+    const int usec = 1000000 / wm_max_eps;
+
+    mtdebug1(WM_SYS_LOGTAG, "Starting ports inventory.");
+
+    pid_t * pids = NULL;
+    int32_t maxproc;
+    size_t len = sizeof(maxproc);
+    sysctlbyname("kern.maxproc", &maxproc, &len, NULL, 0);
+
+    os_calloc(maxproc, 1, pids);
+    int count = proc_listallpids(pids, maxproc);
 
+    int index;
+    for(index = 0; index < count; ++index) {
+        pid_t pid = pids[index];
+        // Figure out the size of the buffer needed to hold the list of open FDs
+        int bufferSize = proc_pidinfo(pid, PROC_PIDLISTFDS, 0, 0, 0);
+        if (bufferSize == -1) {
+            mterror(WM_SYS_LOGTAG, "Unable to get open file handles for %d", pid);
+            continue;
+        }
+
+        // Get the list of open FDs
+        struct proc_fdinfo *procFDInfo;
+        os_malloc(bufferSize, procFDInfo);
+        if (!procFDInfo) {
+            mterror(WM_SYS_LOGTAG, "Out of memory. Unable to allocate buffer with %d bytes", bufferSize);
```

Comment:
This is dead code since `os_malloc` closes the program when the second parameter is `NULL`. If you want to change that behavior and let this code, use the standard `malloc` instead.